# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 11a3c2c86a8e
# Full Hash: 11a3c2c86a8e85fbbfe0fad4d36bf3956af109c5
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 9: Implement alertclickcallback behavior r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227694
# ==============================================================================

diff -r 10bb5f41f44b -r 11a3c2c86a8e dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Mon Nov 11 20:34:14 2024 +0000
+++ b/dom/ipc/ContentParent.cpp	Mon Nov 11 20:34:14 2024 +0000
@@ -4751,20 +4751,12 @@
     return IPC_OK();
   }
 
-  if (aType.EqualsLiteral("click")) {
-    nsresult rv = swm->SendNotificationClickEvent(
-        aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
-        aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
-        aData.data(), aData.behavior());
-    Unused << NS_WARN_IF(NS_FAILED(rv));
-  } else {
-    MOZ_ASSERT(aType.EqualsLiteral("close"));
-    nsresult rv = swm->SendNotificationCloseEvent(
-        aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
-        aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
-        aData.data(), aData.behavior());
-    Unused << NS_WARN_IF(NS_FAILED(rv));
-  }
+  MOZ_ASSERT(aType.EqualsLiteral("close"));
+  nsresult rv = swm->SendNotificationCloseEvent(
+      aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
+      aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
+      aData.data(), aData.behavior());
+  Unused << NS_WARN_IF(NS_FAILED(rv));
 
   return IPC_OK();
 }